src/ModelManager.ts

```diff
@@ -125,96 +153,170 @@ export class ModelManager {
      *
      * If page model does not contain information about current path it performs additional fetch.
      *
-     * @param [config] URL to the data model or configuration object.
      * @fires cq-pagemodel-loaded
      * @return {Promise}
      */
     public initialize<M extends Model>(config?: ModelManagerConfiguration | string): Promise<M> {
-        this.destroy();
+        this.initializeAsync(config);
 
-        let path;
-        let initialModel = null;
+        const { rootModelURL, rootModelPath } = this._modelPaths;
 
-        if (!config || (typeof config === 'string')) {
-            path = config;
-        } else if (config) {
-            path = config.path;
-            this._modelClient = config.modelClient;
-            initialModel = config.model;
+        if (!rootModelURL) {
+            throw new Error('ModelManager.js Cannot initialize without a URL to fetch the root model');
```

Comment:
- do we want to throw errors? what happens then? is anything usable?
- i would simply say `Provide root model path to initialize ModelManager.`

Reply:
This was part of the existing code for initialize. Most of the error conditions are throwing such errors in ModelManager. Should we move away from such patterns?

Reply:
@sharanyavinod eventually yes, that could be ui notification like most of our ui does to report errors; if that was existing move moved around, lets stay wit it for now